libraries/core/src/main/kotlin/com/bumble/appyx/core/store/RetainedInstanceStoreImpl.kt

```diff
@@ -0,0 +1,28 @@
+package com.bumble.appyx.core.store
+
+import kotlin.reflect.KClass
+
+internal class RetainedInstanceStoreImpl : RetainedInstanceStore {
+
+    private val map: MutableMap<String, MutableMap<KClass<*>, ValueHolder<*>>> = HashMap()
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : Any> get(nodeId: String, clazz: KClass<*>, disposer: (T) -> Unit, factory: () -> T): T =
```

Comment:
Use a string key instead of the class so that multiple instance of the same class can be used.